src/mono/mono/metadata/assembly.c

```diff
@@ -969,7 +969,7 @@ mono_assembly_load_reference (MonoImage *image, int index)
 		goto commit_reference;
 	}
 
-	if (image->assembly) {
+	if (image->assembly || image->not_executable) {
```

Comment:
Let's drop the `else` part here.

ie, rewrite as:
```c
g_assertf (image->assembly || image->not_executable, ""While loading reference %d MonoImage %s doesn't have a MonoAssembly", index, image->name);
if (mono_trace_is_traced (G_LOG_LEVEL_INFO, MONO_TRACE_ASSEMBLY)) {
   ...
}
MonoAssemblyByNameRequest req;
...
```

Reply:
Done, also modified message to remark `|| image->not_executable`